### DISCUSSION TITLE
connect EMFILE 52.119.224.234:443 error

### DISCUSSION THREAD

**Discussion by webtwitter** [tinacms/tinacms#3682] [Help] [answered]
Hi, when I'm trying to deploy on Vercel, I randomly get this error:
connect EMFILE 52.119.224.234:443

My stack is:
- next js 13.2.3 using app 
- tinacms 1.3.0 (tried with earlier versions but still the same)

Site is quite simplistic for now: https://aimania.one/ 

As I told you, I'm not sure if this issue is from my side since it appears random.
Is there something I can add in tina config file to also optimize this? It also builds 18 pages on 30 seconds for static sites (if it's from Vercel, that's fine though)

Thanks a lot!

![Screenshot_1](https://user-images.githubusercontent.com/32715017/223921280-f27a6959-e833-4976-a3d8-fcb36eaed8b7.jpg)

**Comment by webtwitter** [comment]
Anyone else has this error? Or is anyone able to help?
It's a nightmare to deploy a change for this website having multiple errors like this. Only on rare occasions, it works (randomly).

**Reply by kldavis4** [reply]
This SO answer might help: https://stackoverflow.com/a/32365535/290918

**Reply by webtwitter** [reply]
@kldavis4 thanks a lot for your answer, but since I have no control over any server, it seems like I can't do anything here.
I just use what tina.io docs suggest. And for 77 pages it crashes.
It seems like the IP is from Tina hosting AWS actually, so  I have a feeling that I have no control over this error.

**Reply by kldavis4** [reply]
Please make a reproduction and file an issue and we can investigate further.

**Reply by webtwitter** [reply]
@kldavis4  I'm also changing the code for static pages to read with fetch() https://tina.io/docs/reference/content-api/content-delivery/ I'll let you know if this will change things and maybe create a use case if this is helpful.

**Comment by webtwitter** [comment, marked as answer]
Update on this:
Because the problem still persisted I decided to use tina's content api https://tina.io/docs/reference/content-api/content-delivery/
After I'm using this approach it works like a charm, because:
- I'm modelling the content the way I want
- I can do custom / multiple requests
- I'm also caching the API

So thank you for also letting us use other methods :)
From 3+minutes of building using Tina's library, to <1min builds using custom API :D
@kldavis4

**Reply by cascading-jox** [reply]
Would you mind sharing the code for your custom client? I have the exact same issue!
